calculator.c

```diff
@@ -366,27 +369,33 @@ Symbol tokenType(token tk)
 		case digit:
 			ret = value;
 			break;
+		default:
+			break;
 	}
 	return ret;
 }
 
-token tokenizeNumber(char *str)
-{
-	;
-}
 
 int tokenize(char *str, char *(**tokensRef))
 {
 	char** tokens = NULL;
+	char** tmp = NULL;
 	char* ptr = str;
 	char ch = '\0';
 	int numTokens = 0;
-	while(ch = *ptr++)
+	char* tmpToken = malloc((MAXTOKENLENGTH+1) * sizeof(char));
+	if (!tmpToken)
+	{
+		fprintf(stderr, "Malloc of temporary buffer failed\n");
+		return 0;
+	}
+	while((ch = *ptr++))
 	{
 		if(type(ch) == invalid) // Stop tokenizing when we encounter an invalid character
 			break;
 
 		token newToken = NULL;
+		memset(tmpToken, '\0', MAXTOKENLENGTH+1);
```

Comment:
I don't think that it harms performance to an extend that the calculator seems slow. I do the memset only to have a consistent state at each tokenization step. It is probably enough to set only the first entry to a null byte for the `if` you mentioned.